konnectivity-socks5-proxy/main.go

```diff
@@ -223,6 +276,13 @@ func (d proxyResolver) Resolve(ctx context.Context, name string) (context.Contex
 		address, err := d.guestClusterResolver.resolve(ctx, name)
 		if err != nil {
 			l.Error(err, "failed to look up address from guest cluster")
+
+			if d.resolveFromManagementCluster {
+				l.Info("Fallback to management cluster resolution")
+				d.dnsFallback.set(true)
+				return ctx, nil, nil
+			}
+
 			return ctx, nil, fmt.Errorf("failed to look up name %s from guest cluster cluster-dns: %w", name, err)
```

Comment:
If resolution succeeds, then change it to false
} else {
    if d.resolveFromManagementCluster {
        d.dnsFallback.set(false) 
    }
}